demisto_sdk/dev_tools/initiator.py

```diff
@@ -0,0 +1,286 @@
+    """
+
+    def __init__(self, output_dir: str, name: str, integration: bool = False, script: bool = False):
+        if output_dir:
+            self.output_dir = output_dir
+
+        else:
+            self.output_dir = ''
+
+        self.name = name
+        self.integration = integration
+        self.script = script
+        self.full_path = ''
+
+    def init(self):
+        """Starts the init command process.
+
+        """
+        if self.integration:
+            self.check_name(created_object="integration")
+            self.integration_init()
+
+        elif self.script:
+            self.check_name(created_object="script")
+            self.script_init()
+
+        else:
+            self.check_name(created_object="pack")
+            self.pack_init()
+
+    def check_name(self, created_object: str):
+        """Makes sure a name is given for the created object
+
+        Args:
+            created_object (str): the type of the created object (integration/script/pack)
+        """
+        while self.name is None:
+            print(f"Please input the name of the initialized {created_object}:")
+            self.name = str(input())
+
+    def pack_init(self) -> bool:
+        """Creates a pack directory tree.
+
+        Returns:
+            bool. Returns True if pack was created successfully and False otherwise
+        """
+        # if an output directory given create the pack there
+        if len(self.output_dir) > 0:
+            self.full_path = self.output_dir + '/' + self.name
+
+        # content-descriptor file indicates we are in "content" repository
+        # thus we will create the pack under Packs directory
+        elif os.path.isfile('content-descriptor.json'):
+            self.full_path = "Packs/" + self.name
+
+        # if non of the above conditions apply - create the pack in current directory
+        else:
+            self.full_path = self.name
+
+        if not self.create_new_directory():
+            return False
+
+        os.mkdir(self.full_path + INCIDENT_FIELDS_DIR)
+        os.mkdir(self.full_path + INDICATOR_FIELDS_DIR)
+        os.mkdir(self.full_path + CLASSIFIERS_DIR)
+        os.mkdir(self.full_path + CONNECTIONS_DIR)
+        os.mkdir(self.full_path + DASHBOARDS_DIR)
+        os.mkdir(self.full_path + INCIDENT_TYPES_DIR)
+        os.mkdir(self.full_path + MISC_DIR)
+        os.mkdir(self.full_path + REPORTS_DIR)
+        os.mkdir(self.full_path + WIDGETS_DIR)
+        os.mkdir(self.full_path + INTEGRATIONS_DIR)
+        os.mkdir(self.full_path + LAYOUTS_DIR)
+        os.mkdir(self.full_path + PLAYBOOKS_DIR)
+        os.mkdir(self.full_path + SCRIPT_DIR)
+        os.mkdir(self.full_path + TEST_PLAYBOOKS_DIR)
+
+        with open(self.full_path + '/CHANGELOG.md', 'a') as fp:
+            fp.write("## [Unreleased]")
+
+        fp = open(self.full_path + '/README.md', 'a')
+        fp.close()
+
+        with open(self.full_path + '/pack-metadata.json', 'a') as fp:
+            fp.write('[]')
```

Comment:
add TODO note to fill this in while the metadata.json script will be ready

Reply:
done